// webdriverio/pages/Home.js
'use strict';

var chai = require('chai');
var expect = chai.expect;
var config = require('../config');

module.exports = function(client, baseURL, platform) {
  this.post_button = "[data-qa=post-button]";
  this.myolx = "[data-qa=my-olx]";
  this.logout_button = "[data-qa=logout-link]";
  this.select_city_link = "[data-qa=select-city]";
  this.change_city_link = "[data-qa=change-city]";
  this.search_field = "[data-qa=search-input]";
  this.search_button = "[data-qa=search-submit]";
  
  this.go = function() {
        client
            .timeouts('implicit', config.timeout)
            .url(baseURL + '/?location=www.olx.com.py')
            .deleteCookie()
            .setCookie({name: 'forcedPlatform', value: platform })
            .setCookie({name: 'showInterstitial', value: '1' })
            .refresh();
    };
  

  this.goToPostingPage = function() {
    client
        .click(this.post_button);
    };

  this.goToLoginPage = function() {
     client
        .click(this.myolx);
    };


  this.logOut = function(){
    client
        .click(this.myolx)
        .waitForVisible(this.logout_button)
        .click(this.logout_button);
    };

  this.isUserLoggedOut = function(){
      var myolx = this.myolx;
      var user_logged_out = "[href*='/login']";
      client
          .isExisting(user_logged_out, function(err, isExisting) {
            expect(isExisting).to.equal(true);
      });
  };


  this.isUserLoggedIn = function(username, password) {
      var myolx = this.myolx;
      var user_logged_out = "[href*='/login']";
      client
          .isExisting(user_logged_out, function(err, isExisting) {
            expect(isExisting).to.equal(false);
      });
  };



  this.goToChangeCity = function(){
     client
          .click(this.change_city_link);     
  };

  this.goToSelectCity = function(){
     client
          .click(this.select_city_link);     
  };

  this.isUserLocatedInCity = function() {
    var change_city = this.change_city_link;
    client
        .isExisting(change_city, function(err, isExisting) {
        expect(isExisting).to.equal(true);
    });
  };

  this.globalSearch = function(term){
    client
        .setValue(this.search_field, term)
        .click(this.search_button);
  };

}
