// webdriverio/pages/Item.js
'use strict';

var chai = require('chai');
var expect = chai.expect;
var config = require('../config');

module.exports = function(client, baseURL, platform) {
  if(platform == 'html4'){
    this.remove_favorite = "[data-qa=remove-favorite]"; 
    this.add_favorite = "[data-qa=add-favorite]";
  }
    else{
      this.remove_favorite = "[class*='fav'][class*='remove']"; 
      this.add_favorite = "[class*='fav'][class*='add']";
    }

/*
  this.addItemToFavorites = function(){
    var remove_favorite = this.remove_favorite;
    var add_favorite = this.add_favorite;
    driver.manage().timeouts().implicitlyWait(0); 
    driver.isElementPresent(remove_favorite)
      .then(function check(isPresent) {
        if (isPresent){
          driver.findElement(remove_favorite).click();
        }
      })
      .then(function click() {
        driver.manage().timeouts().implicitlyWait(config.timeout);  
        driver.findElement(add_favorite).click();
        driver.manage().timeouts().implicitlyWait(0);  
      })
      .then(function check() {
        driver.isElementPresent(remove_favorite)
          .then(function assert(isPresent) {
            expect(isPresent).to.equal(true);
          });
      });
    driver.manage().timeouts().implicitlyWait(config.timeout);   
  };
*/


  this.isItemDisplayed = function(){
      var item_page_element = "[data-qa=item]";
      client
          .isExisting(item_page_element, function(err, isExisting) {
            expect(isExisting).to.equal(true);
      });
  };       
}
